File: frontend/olive-platform/src/features/olivePurchases/data/responses/OlivePurchaseItemResponse.ts

export type OlivePurchaseItemResponse = {
  id: number

  reference: string

  purchaseId: number

  varietyId: number | null

  description: string | null

  agreedQuantityKg: number

  pressedQuantityKg: number

  remainingQuantityKg: number

  pricePerKg: number

  totalAmount: number | null

  notes: string | null
}

File: backend/OlivePlatform/OlivePlatform.Application/Features/OlivePurchases/Responses/OlivePurchaseItemResponse.cs
namespace OlivePlatform.Application.Features.OlivePurchases.Responses
{
    public class OlivePurchaseItemResponse
    {
        public int Id { get; set; }

        public required string Reference { get; set; }

        public int? VarietyId { get; set; }

        public string? VarietyName { get; set; }

        public decimal AgreedQuantityKg { get; set; }

        public decimal RemainingQuantityKg { get; set; }

        public decimal PricePerKg { get; set; }

        public int TotalAmount { get; set; }
    }
}
